pkg/runtime/function/acos.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package function
+
+import (
+	"context"
+	"math"
+)
+
+import (
+	"github.com/pkg/errors"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+)
+
+// FuncAcos https://dev.mysql.com/doc/refman/5.6/en/mathematical-functions.html#function_acos
+const FuncAcos = "ACOS"
+
+var _ proto.Func = (*acosFunc)(nil)
+
+func init() {
+	proto.RegisterFunc(FuncAcos, acosFunc{})
+}
+
+type acosFunc struct{}
+
+// Apply call the current function.
+func (s acosFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	param, err := inputs[0].Value(ctx)
+	if param == nil || err != nil {
+		return nil, errors.WithStack(err)
+	}
```

Comment:
<img width="477" alt="image" src="https://user-images.githubusercontent.com/1284892/213070548-467de3ad-6db8-4c94-be3d-aa9e355348bf.png">
When `param` here is nil, should return `nil,nil`.

PS: please check the rest of functions in this PR, there're same problem.

Reply:
We consider two cases:

- If `err != nil`, the function will return `nil, errors.WithStack(err)`  as before.
- If `param == nil && err == nil`, the function will return `nil, nil` as before too.  This is because `errors.WithStack` return `nil`, if input is `nil`.

![withstack](https://user-images.githubusercontent.com/16648345/213090453-421bdd08-08a5-485e-8254-9e0828488fef.png)

I add more test cases for these math functions, which includes the `nil` input.

Reply:
Oops... it seems right. 🤣